scripts/multicluster/setup.sh

```diff
@@ -0,0 +1,176 @@
+    kubectl apply -f https://raw.githubusercontent.com/openservicemesh/osm-docs/release-v1.2/manifests/apps/bookstore.yaml
+    kubectl apply -f https://raw.githubusercontent.com/openservicemesh/osm-docs/release-v1.2/manifests/apps/bookwarehouse.yaml
+    kubectl apply -f https://raw.githubusercontent.com/openservicemesh/osm-docs/release-v1.2/manifests/apps/mysql.yaml
+done
+
+# ==== Deploy a curl client for testing
+${k1} apply -f - <<EOF
+apiVersion: v1
+kind: Deployment
+metadata:
+  name: curl
+  namespace: bookbuyer
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: curl
+  template:
+    metadata:
+      labels:
+        app: curl
+    spec:
+      containers:
+      - name: curl
+        image: curlimages/curl
+        command: ["sleep", "1000000"]
+EOF
+
+# ==== 3. Create ServiceExport
+${k2} apply -f - <<EOF
+apiVersion: multicluster.x-k8s.io/v1alpha1
+kind: ServiceExport
+metadata:
+  name: bookstore
+  namespace: bookstore
+EOF
+
+# ==== 4. Create ServiceImport
+
+# create a meta-service to get a valid local service IP
+${k1} apply -f - <<EOF
+apiVersion: v1
+kind: Service
+metadata:
+  name: mcs-bookstore
+  namespace: bookstore
+spec:
+  type: ClusterIP
+  ports:
+  - port: ${app_port}
+    protocol: TCP
+EOF
+sleep 5
+mcs_ip=$(${k1} get service mcs-bookstore -n bookstore -o jsonpath='{.spec.clusterIP}')
+
+${k1} apply -f - <<EOF
+apiVersion: multicluster.x-k8s.io/v1alpha1
+kind: ServiceImport
+metadata:
+  name: bookstore
+  namespace: bookstore
+spec:
+  ips:
+  - ${mcs_ip}
+  type: ClusterSetIP
+  ports:
+  - name: http
+    protocol: TCP
+    port: ${app_port}
+EOF
+
+# ==== 5. Expose service on c2 and get the IP addresses.
+${k2} patch service bookstore -n bookstore -p '{"spec":{"type":"LoadBalancer"}}'
+for _ in {1..10}; do
+    remote_ip=$(${k2} get service bookstore -n bookstore -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+
+    if [ -n "${remote_ip}" ]; then
+        break
+    fi
+    sleep 5
+done
+if [ -z "${remote_ip}" ]; then
+    echo "Remote bookstore IP not found"
+    exit 1
+fi
+echo "Remote bookstore IP: ${remote_ip}"
+
+local_ip=$(${k1} get service bookstore -n bookstore -o jsonpath='{.spec.clusterIP}')
+
+# ==== 6. Create remote endpointSlice
+uid=$(${k1} get serviceimport bookstore -n bookstore -o jsonpath='{.metadata.uid}')
+${k1} apply -f - <<EOF
+apiVersion: discovery.k8s.io/v1
+kind: EndpointSlice
+metadata:
+  name: imported-bookstore-cluster-c2
+  namespace: bookstore
+  labels:
+    multicluster.kubernetes.io/source-cluster: cluster-c2
+    multicluster.kubernetes.io/service-name: bookstore
+  ownerReferences:
+  - apiVersion: multicluster.k8s.io/v1alpha1
+    controller: false
+    kind: ServiceImport
+    name: bookstore
+    uid: ${uid}
+addressType: IPv4
+ports:
+  - name: http
+    protocol: TCP
+    port: ${app_port}
+endpoints:
+  - addresses:
+    - "${remote_ip}"
```

Comment:
This should be a list of actual pod IPs

Reply:
The actual pod IPs needs the assumption that two clusters are in the flat network (this applies to the last comment also). If they are not in a flat network, AFAIK an extra layer (like Submariner which provides cluster gateway) is required. In our simple setup, we don't have these.

Reply:
agreed with @allenlsy  in a non-flat network, we will want this field here.

We may add an OSM object that will enable this, by putting the remote IP in a separate location, and we use that instead, TBD.

For now I think this is acceptable, although maybe we should add a comment about that

Reply:
Yeah this is likely fine for our setup...but there's some merging with duplicate serviceexport names in the original spec that I want to make sure we're aware of

Reply:
can you elaborate?

Reply:
From [the KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-multicluster/1645-multi-cluster-services-api)

> If multiple clusters export a Service with the same namespaced name, they will be recognized as a single combined service. For example, if 5 clusters export my-svc.my-ns, each importing cluster will have one ServiceImport named my-svc in the my-ns namespace and it will be associated with endpoints from all exporting clusters. Properties of the ServiceImport (e.g. ports, topology) will be derived from a merger of component Service properties.

I don't see anything that requires a merge between a commonly named local Service and ServiceImports, but if the owning cluster creates a ServiceExport for that internal service, then this seems to imply that the endpoints should be merged. The non-flat network question may need some clarification in the upstream KEP

Reply:
I think when importing external services, the importing cluster will copy the EndpointSlices from the exporting cluster, but not merging the endpoints to existing EndpointSlice. Hope this clarifies the process.

Either way (using a single remote service IP or a list of remote pod IPs) are reasonable here. It depends on how we create the multiple clusters.